### DISCUSSION TITLE
VLC VideoView in Avalonia Linux FrameBuffer

### DISCUSSION THREAD

**Discussion by kipy-be** [AvaloniaUI/Avalonia#15105] [Q&A]
Hello, I've tested the Avalonia DRM linux mode on a raspberry pi and it works like a charm !

I then wanted to test the VideoView from the package LibVLCSharp.Avalonia, added it into the main view, and set ip up to load a video after the component was loaded. 

- It works in Windowed mode (windows / rasp with X)
- In DRM mode, the video won't show up in the embedded video view

Did someone tried it before ? Is there sort a trick ? Or it's simply not possible yet ?

Thanks you for you reply

**Comment by thevortexcloud** [comment]
I don't think it's possible? I have not tried it so someone may say otherwise: But the way the VLC integration works as far as I am aware is by embedding the VLC _window_ into Avalonia (you can actually see this is you set things up in the wrong order, in which case a separate window will appear for the video). Since there is no windowing system in DRM mode it's not possible to do it. If you could get VLC to instead render to a texture/some other output, you could probably get it to work another way. If you can get it to output to a bitmap, you could probably get it working via a `RenderTargetBitmap`, although it probably wont be very fast if you go via that route. VLC does actually have a Framebuffer mode, but you likely wont be able to use it and Avalonia at the same time.

It may be possible (I have not looked into it at all) to also get the OpenGL texture from VLC when it's in OpenGL mode and use that in the OpenGL control. But I have no idea if that works with the DRM backend in Avalonia.

**Reply by kipy-be** [reply]
Thank you for your reply !

Indeed, VLC has framebuffer (/dev/fbx) and DRM outputs (/dev/dri/cardx) support but it cannot "share" them with Avalonia easily.
The OpenGL way was indeed on my list, but VLC has currently no possibility to render on a user GL context. (but this is an upcoming VLC 4 feature, apparently).

There is a way to configure DRM output to have severals "planes" (one as primary - the video layer -, and others overlays), then set the color output mode of the overlay layer (ie avalonia) with opacity to be able to blend the two. I'm currently investigating that.

**Reply by javidsho** [reply]
@kipy-be Could you please share how your investigation went?

**Comment by javidsho** [comment]
Any news with this? This would be an amazing feature for embedded software!

**Reply by timunie** [reply]
https://docs.avaloniaui.net/accelerate/components/media-player/quickstart#embedded-linux-direct-rendering-manager
